src/sat/sat_solver.h

```diff
@@ -176,8 +177,7 @@ namespace sat {
         std::string             m_reason_unknown;
         bool                    m_trim = false;
 
-        svector<unsigned>       m_visited;
-        unsigned                m_visited_ts;
+        visit_helper            m_visited;
```

Comment:
instead of exposing this having clienets have to use s().m_visited all over the place have the solver inherit from visit_helper.

Reply:
The reason I want to put this in a class is that I will need two instances of it. (I could also inherit from it in the xor-solver but then I have the problem again that I limit this datastructure to a single solver.)